### ISSUE TITLE
errored: unhandled buildvar type: uint64

### ISSUE THREAD

**Issue by vheon** [hashicorp/packer#9583]
#### Overview of the Issue

I'm playing around with HCL2 syntax for packer and I get an error when I have the `provisioner` block as shown below. As soon as I remove the `provisioner` block it all goes fine. Looking at the error I get `unhandled buildvar type: uint64` it looks related to #9534 but I'm not familiar with the codebase to know for sure, to check or to fix it.

#### Reproduction Steps

this is the `build.pkr.hcl`

```hcl
variable "guest_additions_url" {
  type    = string
  default = ""
}

variable "headless" {
  type    = bool
  default = true
}

locals {
  // fileset lists all files in the http directory as a set, we convert that
  // set to a list of strings and we then take the directory of the first
  // value. This validates that the http directory exists even before starting
  // any builder/provisioner.
  http_directory = dirname(convert(fileset(".", "http/7/*"), list(string))[0])

  iso_url_centos_7 = "http://mirrors.kernel.org/centos/7.8.2003/isos/x86_64/CentOS-7-x86_64-DVD-2003.iso"
  iso_checksum_url_centos_7 = "https://mirrors.edge.kernel.org/centos/7.8.2003/isos/x86_64/sha256sum.txt"
  ks_path = "ks.cfg"
}

source "virtualbox-iso" "base-centos7-amd64" {
  headless = true
  guest_os_type = "RedHat_64"
  http_directory = local.http_directory
  shutdown_command = "echo 'vagrant' | sudo -S /sbin/halt -h -p"
  ssh_username = "vagrant"
  ssh_password = "vagrant"
  ssh_port = 22
  ssh_wait_timeout = "15m"
  hard_drive_interface = "sata"
  virtualbox_version_file = ".vbox_version"
  guest_additions_path = "VBoxGuestAdditions_{{.Version}}.iso"

  boot_command = [
    "<up><wait><tab> text ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/${local.ks_path}<enter><wait>"
  ]
  boot_wait = "10s"

  iso_checksum = "file:${local.iso_checksum_url_centos_7}"
  iso_url = local.iso_url_centos_7

  output_directory = "virtualbox_iso_centos_7_amd64"
}

build {
  name = "centos7"
  description = "build centos7 using virtualbox-iso"

  sources = [ "source.virtualbox-iso.base-centos7-amd64" ]

  provisioner "shell" {
    execute_command = "echo 'vagrant' | {{.Vars}} sudo -S -E sh -eux '{{.Path}}'"
    expect_disconnect = true
    inline = [ "echo 'foo'" ]
  }
}
```

### Packer version

Packer v1.6.1-dev

### Operating system and Environment details

macOS 10.14.6
VirtualBox 6.1.8

### Log Fragments and crash.log files

https://gist.github.com/vheon/143450995eb4b043d61373f08659befa

**Comment by SwampDragons** [comment]
I think this was solved in #9585

**Comment by vheon** [comment]
> I think this was solved in #9585

Yes it was, I just tried it 👍
